core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -337,6 +382,49 @@ private[log] class LogCleanerManager(val logDirs: Seq[File],
       }
     }
   }
+
+  def markPartitionUncleanable(logDir: String, partition: TopicPartition): Unit = {
+    inLock(lock) {
+      uncleanablePartitions.get(logDir) match {
+        case Some(partitions) =>
+          partitions.add(partition)
+        case None =>
+          uncleanablePartitions.put(logDir, mutable.Set(partition))
+      }
+    }
+  }
+
+  def markDirectoryUncleanable(logDir: String): Unit = {
+    inLock(lock) {
+      if (uncleanableDirs.contains(logDir))
+        throw new IllegalStateException(s"Directory $logDir is already marked as uncleanable!")
+
+      uncleanableDirs.add(logDir)
```

Comment:
Do we still need this method? I couldn't find anywhere the state from `uncleanableDirs` is being used other than in `uncleanableDirectories`.

Reply:
No, we don't. Nice catch!